### ISSUE TITLE
Ideas for using fontmake with Noto

### ISSUE THREAD

**Issue by jbmorizot** [googlefonts/noto-source#206]
In order to give users more flexibility, it could be interesting to add more options to fontmake. Some useful options could be:

- Pair fontmake with scale_ufo.py or scale_font.py to output 2000 UPM fonts from 1000 UPM sources;

- Add multiple combining options to fontmake CLI to be able to:
    1. merge writing systems (eg. Noto Naskh Arabic + Noto Sans Latin)
    2. megamerge everything sans into 1 sans family (non cjk)
    3. subset character sets (eg. Noto Latin without IPA and Cyrillic and Greek)
    4. rename binary fonts with custom font names
    5. build Variable Font (possibly slim range, with fontTools.varLib.instancer)
    6. add secure set (currencies, punctuations, basic Latin, figures…)
    7. swap tab and prop figures
    8. swap I J alt
    9. and finally build Static Fonts with some of this previous options

**Comment by davelab6** [comment]
@jbmorizot please could you summarize the conclusions of the call today where this was discussed? :)

**Comment by jbmorizot** [comment]
It appears that the best way to handle different fonts building (merged, renamed, subsetted, etc.) is to plug existing tools and libraries, and not update fontmake for these specific uses.
That said, to deal with the needs of Noto, the following things could be done:
• Scale_ufo/scale_fonts could be implemented in ufo2ft.
• The best way to handle merging may be at binary level, with the fontTools merger (that could be improved, and is the most interresting tool to build upon).
• Same goes for the megamerge, subsetting, renaming.
• If merging has to be done at sources (= ufo) level, the right way would be to remove duplicated glyphs by using objects provided by ElementTree for updating the .fea file.
One important point is that merging takes longer that subsetting. So a good workaround could be to have a (several) megamerged font(s) and subset it (them) on demand to provide custom fonts.
